### DISCUSSION TITLE
Support for request examples?

### DISCUSSION THREAD

**Discussion by vishalbd** [scalar/scalar#764] [Ideas]
Hi ✋ 

I was trying to setup Scalar for my OpenAPI schema, but noticed that there was no way to view examples that I have provided for the request schema. The generated sample request body for both the preview and the curl and try-it-out console are populated with default values from the schema, but ideally we would have an option to select from one of the provided examples...

I find it a little odd because the example for the response schema is properly displayed. I'm just wondering if it is not implemented in the library yet or if I am missing something with the docs or configuration. Any pointers would be greatly appreciated!

**Comment by marclave** [comment]
hi @vishalbd 👋 

would you mind sharing your openapi specification or swagger file, if not it would be very helpful to have a snippet if possible :) 

if you can't share the snippet, could you provide more info on what fields you have set and that aren't being rendered ✨ hopefully its something we can fix quick :)

**Comment by vishalbalaji** [comment]
Hey @marclave! Here is a gist for a sample OpenAPI schema, where both the request and response schemas themselves have examples:
https://gist.github.com/vishalbalaji/81c02b08ab3683a87813926466633c48

A screenshot from the Scalar demo on how this is rendered:
![image](https://github.com/scalar/scalar/assets/36506250/5069fbf9-0d2d-4f7a-9e09-872529ed0339)

As can be seen here, The response example is rendered properly, but the request example is nowhere to be seen. Here, the curl also generates the data as `null` for some reason, which hasn't happened with me for other schemas. But usually, it just generates a sample example with the default values for each attribute, like in the petstore example.

For reference, here's how Stoplight Elements renders it:
![image](https://github.com/scalar/scalar/assets/36506250/bf1f4bbb-9c60-4297-b57e-a1b3c4ae5d76)
Live preview: https://elements-demo.stoplight.io/?spec=https%3A%2F%2Fgist.github.com%2Fvishalbalaji%2F81c02b08ab3683a87813926466633c48%2Fraw%2Fbd23008789af6fec14f7eab091e225300959b793%2Fsample-openapi-schema.json#/operations/1

Here, we are able to select between examples if multiple of them are specified and it even automatically updates the code/request sample, which is super convenient when testing out and endpoint.

**Reply by hanspagel** [reply]
Thanks for taking the time to bring this up! I made a PR. :)

https://github.com/scalar/scalar/pull/811
